### ISSUE TITLE
semver {} config block does not work

### ISSUE THREAD

**Issue by javaboy79** [jmongard/Git.SemVersioning.Gradle#57]
since  0.12.0 the semver {} config block does not seem to work.  I tested with the following and the settings had no effect.  0.11.0 works.

```
semver {
	defaultPreRelease = "PRE"
	noDirtyCheck = true
}

version = semver.version
```

**Comment by jmongard** [comment]
Yes, you are right. Making the plugin compatible with configuration cache is hard. I guess I need to figure out how to handle the settings correctly.

**Comment by jmongard** [comment]
With the introduction of the configuration cache  the semver properties were read before you had the chance to set their value.
I have fix this in version `0.12.6`
